amoro-format-iceberg/src/main/java/org/apache/amoro/optimizing/plan/CommonPartitionEvaluator.java

```diff
@@ -208,6 +217,11 @@ private boolean addTargetSizeReachedFile(DataFile dataFile, List<ContentFile<?>>
     return false;
   }
 
+  private void updateFileSizeSquaredErrorSum(DataFile dataFile) {
+    long diffSize = minTargetSize - Math.min(dataFile.fileSizeInBytes(), minTargetSize);
```

Comment:
maybe add comments that only accumulates squared error for files smaller than minTargetSize.